File: All/class01/1.cpp
// 利用前缀和快速得到区域累加和
// 测试链接 : https://leetcode.cn/problems/range-sum-query-immutable/

class NumArray {
public:

    vector<int> arr;
    NumArray(vector<int>& nums) {
        arr.resize(nums.size()+1);
        for(int i=1;i<=nums.size();i++){
            arr[i] = nums[i-1]+arr[i-1];
        }
    }
    
    int sumRange(int left, int right) {
        return arr[right+1]-arr[left];
    }
};

File: All/class01/1.py
# 利用前缀和快速得到区域累加和
# 测试链接 : https://leetcode.cn/problems/range-sum-query-immutable/

class NumArray:

    def __init__(self, nums: List[int]):
        self.arr = [0 for i in range(len(nums)+1)]

        for i in range(1,len(nums)+1):
            self.arr[i] = nums[i-1]+self.arr[i-1]

    def sumRange(self, left: int, right: int) -> int:
        return self.arr[right+1]-self.arr[left]
